src/modules/esl-carousel/core/nav/esl-carousel.nav.utils.ts

```diff
@@ -0,0 +1,123 @@
+function parseIndex(index: string | ESLCarouselNavIndex): {value: number, isRelative: boolean, dir?: ESLCarouselDirection} {
+  if (typeof index === 'number') return {value: index, isRelative: false};
+  index = index.trim();
+  if (index === 'next') return {value: 1, isRelative: true, dir: 'next'};
+  if (index === 'prev') return {value: -1, isRelative: true, dir: 'prev'};
+  if (index[0] === '+') return {value: +index, isRelative: true, dir: 'next'};
+  if (index[0] === '-') return {value: +index, isRelative: true, dir: 'prev'};
+  return {value: +index, isRelative: false};
+}
+
+/** @returns normalized numeric index from string with absolute or relative index */
+function resolveSlideIndex(indexStr: string | ESLCarouselNavIndex, cfg: ESLCarouselState): {index: number, dir: ESLCarouselDirection | null} {
+  const {value, isRelative, dir} = parseIndex(indexStr);
+  const target = value + (isRelative ? cfg.activeIndex : 0);
+  const index = cfg.loop ? normalizeIndex(target, cfg.size) : boundIndex(target, cfg.size - cfg.count);
+  return {index, dir: dir || indexToDirection(index, cfg)};
+}
+
+/** @returns normalized numeric index from string with absolute or relative group index */
+function resolveGroupIndex(indexStr: string | ESLCarouselNavIndex, cfg: ESLCarouselState): {index: number, dir: ESLCarouselDirection | null} {
+  const {value, isRelative, dir} = parseIndex(indexStr);
+  if (!isRelative) {
+    const index = groupToIndex(value, cfg.count, cfg.size);
+    return {index, dir: indexToDirection(index, cfg)};
+  }
+  // TODO: extend navigation boundaries
+  if (value === -1 && cfg.activeIndex < cfg.count && cfg.activeIndex > 0) {
+    return {index: 0, dir: dir || 'prev'};
+  }
+  if (value === 1 && normalizeIndex(cfg.activeIndex + cfg.count, cfg.size) > cfg.size - cfg.count) {
+    return {index: cfg.size - cfg.count, dir: dir || 'next'};
+  }
+  const index = normalizeIndex(cfg.activeIndex + value * cfg.count, cfg.size);
+  return {index, dir: dir || indexToDirection(index, cfg)};
+}
+
+/** @returns normalized index from target definition and current state */
+export function toIndex(target: ESLCarouselSlideTarget, cfg: ESLCarouselState): {index: number, dir: ESLCarouselDirection | null} {
+  if (typeof target === 'number') {
+    const index = cfg.loop ? normalizeIndex(target, cfg.size) : boundIndex(target, cfg.size - cfg.count);
+    return {index, dir: indexToDirection(index, cfg)};
+  }
+  const {type, index} = splitTarget(target);
+  if (type === 'group') return resolveGroupIndex(index, cfg);
+  if (type === 'slide') return resolveSlideIndex(index, cfg);
+  return {index: cfg.activeIndex, dir: null};
+}
+
+export function canNavigate(target: ESLCarouselSlideTarget, cfg: ESLCarouselState): boolean {
+  const {dir, index} = toIndex(target, cfg);
+  if (!cfg.loop && index > cfg.activeIndex && dir === 'prev') return false;
+  if (!cfg.loop && index < cfg.activeIndex && dir === 'next') return false;
```

Comment:
```suggestion
  if (!cfg.loop && index !== cfg.activeIndex && ['prev', 'next'].includes(dir)) return false;
```

Reply:
That is not equal condition

`(index > cfg.activeIndex && dir === 'prev') || (index < cfg.activeIndex && dir === 'next')`  
!<==> 
`index !== cfg.activeIndex && ['prev', 'next'].includes(dir)`